certbot-nginx/certbot_nginx/parser.py

```diff
@@ -549,15 +530,36 @@ def _add_directive(block, directive, replace):
         # Append directive. Fail if the name is not a repeatable directive name,
         # and there is already a copy of that directive with a different value
         # in the config file.
+
+
+        # handle flat include files
+
         directive_name = directive[0]
-        if location is None or (isinstance(directive_name, str) and
-                                directive_name in REPEATABLE_DIRECTIVES):
+        can_append = lambda loc, dir_name: loc is None or (isinstance(dir_name, str) \
+                                and dir_name in REPEATABLE_DIRECTIVES)
+
+        cannot_append_error = lambda direc, loc: errors.MisconfigurationError( \
```

Comment:
nit: This is a somewhat strange use of a lambda in my opinion. Maybe instead you should make an `ERR_FMT` string that you can use or a simple function called something like `_fail_add_directive()`.